ripd/rip_nb_config.c

```diff
@@ -964,15 +972,97 @@ int lib_interface_rip_authentication_password_modify(
 {
 	struct interface *ifp;
 	struct rip_interface *ri;
+	const char *password_in;
+	bool is_encrypted = false;
+	size_t pwlen;
+	struct tmp_rip_auth_password_strings *pwstrs;
+
+	password_in = yang_dnode_get_string(args->dnode, NULL);
+
+	switch (args->event) {
+	case NB_EV_VALIDATE:
+		if (!strncmp(password_in, "101 ", 4)) { /* hack! */
+			is_encrypted = true;
+			password_in = password_in + 4;
+		}
```

Comment:
> Hard NAK on a semantic level since it is very unclear to me if/how this will interact with YANG config. The missing `isisd` support serves well as canary. Please implement `isisd` support to show how the pieces will interact.

I hadn't even gotten down this far since isisd just jumped out at me, but ... I really don't consider this shippable.